js/style/style.js

```diff
@@ -66,12 +72,36 @@ function Style(stylesheet, animationLoop) {
         this.glyphSource = new GlyphSource(stylesheet.glyphs);
         this._resolve();
         this.fire('load');
+
+        if (validateStyle.emitErrors(this, validateStyle(stylesheet))) return;
```

Comment:
Why did this move? Don't we want to continue to short circuit if there's a validation error?